File: chrome/test/data/webui/chromeos/ash_common/post_message_api/post_message_api_browsertest.js
// Copyright 2021 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

// Polymer BrowserTest fixture.
GEN_INCLUDE(['//chrome/test/data/webui/chromeos/polymer_browser_test_base.js']);

GEN('#include "content/public/test/browser_test.h"');

function PostMessageApiModuleTest() {}

PostMessageApiModuleTest.prototype = {

  __proto__: testing.Test.prototype,

  /** @override */
  browsePreload:
      'chrome://chrome-signin/test_loader.html?module=chromeos/ash_common/post_message_api/post_message_api_test.js',

  isAsync: true,
};

// Test the postmessage api server and client by defining server methods, and
// passing a series of postmessages.
TEST_F('PostMessageApiModuleTest', 'PostMessageCommTest', function() {
  runMochaTest('PostMessageApiModuleTest', 'PostMessageCommTest');
});

File: chrome/test/data/webui/chromeos/ash_common/post_message_api/post_message_api_browsertest.cc
// Copyright 2025 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "chrome/common/webui_url_constants.h"
#include "chrome/test/base/web_ui_mocha_browser_test.h"
#include "content/public/test/browser_test.h"

class PostMessageApiBrowserTest : public WebUIMochaBrowserTest {
 protected:
  PostMessageApiBrowserTest() {
    set_test_loader_host(chrome::kChromeUIChromeSigninHost);
  }
};

IN_PROC_BROWSER_TEST_F(PostMessageApiBrowserTest, AllTests) {
  RunTest("post_message_api/post_message_api_test.js", "mocha.run()");
}
